bot.go

```diff
@@ -47,6 +49,41 @@ func (b Bot) Listen(subscription chan<- Message, timeout time.Duration) {
 	}()
 }
 
+// ListenWebHook starts HTTPS server and register webHooks in Telegram for PUSH
+// updates. certFile and keyFile - filenames containing a certificate and matching private key for the server
+// See https://golang.org/pkg/net/http/#ListenAndServeTLS
+func (b Bot) ListenWebHook(subscription chan<- Message, publicAddress, path, certFile, keyFile string) error {
```

Comment:
We could go for a wrapper then. The current implementation is way too quirky for the API niceness level, telebot aims for.